src/sage/rings/semirings/tropical_polynomial.py

```diff
@@ -0,0 +1,986 @@
+            val_root = dict_coeff[index1] + index1*root
+            check_maks = True
+            for key in dict_coeff:
+                if key not in comb:
+                    val = dict_coeff[key] + key*root
+                    if self.base_ring()._use_min:
+                        if val < val_root:
+                            check_maks = False
+                            break
+                    else:
+                        if val > val_root:
+                            check_maks = False
+                            break
+            if check_maks:
+                order = abs(index1-index2)
+                if root not in dict_root:
+                    dict_root[root] = order
+                else:
+                    if order > dict_root[root]:
+                        dict_root[root] = order
+
+        for root in dict_root:
+            tropical_roots += [root] * dict_root[root]
+
+        return sorted(tropical_roots)
+
+    def split_form(self):
+        r"""
+        Return the tropical polynomial which has the same roots as ``self``
+        but which can be reduced to its linear factors.
+
+        If a tropical polynomial has roots at `x_1, x_2, \ldots, x_n`, then
+        its split form is the tropical product of linear terms of the form
+        `(x + x_i)` for all `i=1,2,\ldots,n`.
+
+        OUTPUT: new :class:`TropicalPolynomial`
+
+        EXAMPLES::
+
+            sage: T = TropicalSemiring(QQ, use_min=True)
+            sage: R.<x> = PolynomialRing(T)
+            sage: p1 = R([5,4,1,0,2,4,3]); p1
+            3*x^6 + 4*x^5 + 2*x^4 + 0*x^3 + 1*x^2 + 4*x + 5
+            sage: p1.split_form()
+            3*x^6 + 2*x^5 + 1*x^4 + 0*x^3 + 1*x^2 + 3*x + 5
+
+        ::
+
+            sage: T = TropicalSemiring(QQ, use_min=False)
+            sage: R.<x> = PolynomialRing(T)
+            sage: p1 = R([5,4,1,0,2,4,3])
+            sage: p1.split_form()
+            3*x^6 + 4*x^5 + 21/5*x^4 + 22/5*x^3 + 23/5*x^2 + 24/5*x + 5
+
+        TESTS:
+
+        Checking that the roots of tropical polynomial and its split form
+        is really the same::
+
+            sage: T = TropicalSemiring(QQ, use_min=True)
+            sage: R.<x> = PolynomialRing(T)
+            sage: p1 = R([5,4,1,0,2,4,3])
+            sage: p1.roots() == p1.split_form().roots()
+            True
+        """
+        roots = self.roots()
+        R = self.parent()
+        poly = R(self.dict()[self.degree()].lift())
+        for root in roots:
+            linear = R([root, 0])
+            poly *= linear
+        return poly
+
+    def factor(self):
+        r"""
+        Return the factorization of ``self`` into its tropical linear factors.
+
+        Note that the factor `x - x_0` in classical algebra gets transformed
+        to the factor `x + x_0`, since the root of the tropical polynomial
+        `x + x_0` is `x_0` and not `-x_0`. However, similar to classical
+        algebra, not every tropical polynomial can be factored.
```

Comment:
```suggestion
        `x + x_0` is `x_0` and not `-x_0`. However, not every
        tropical polynomial can be factored.
```
In classical algebra, every polynomial can be factored if we extend the base ring; this is not the case here.